addons/pos_mercury/static/src/js/PaymentScreen.js

```diff
@@ -254,16 +255,8 @@ odoo.define('pos_mercury.PaymentScreen', function (require) {
                     });
                 }
 
-                this.rpc(
-                    {
-                        model: 'pos_mercury.mercury_transaction',
-                        method: 'do_payment',
-                        args: [transaction],
-                    },
-                    {
-                        timeout: self.server_timeout_in_ms,
-                    }
-                )
+                this.env.services.orm.withOptions({ xhr: makeXHRWithTimeout(self.server_timeout_in_ms) })
+                    .call('pos_mercury.mercury_transaction', 'do_payment', [transaction])
                     .then(function (data) {
```

Comment:
await try/catch? 
then again, this is pos_mercury, dunno if it's still/really used